Microsoft.Toolkit.Uwp.UI.Controls/UniformGrid/UniformGrid.Properties.cs

```diff
@@ -0,0 +1,236 @@
+        public static void SetAutoLayout(FrameworkElement element, bool? value)
+        {
+            element.SetValue(AutoLayoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="FrameworkElement"/></param>
+        /// <returns>A true value indicates this item should be automatically arranged.</returns>
+        public static bool? GetAutoLayout(FrameworkElement element)
+        {
+            return (bool?)element.GetValue(AutoLayoutProperty);
+        }
+
+        /// <summary>
+        /// Sets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="ColumnDefinition"/></param>
+        /// <param name="value">A true value indicates this item should be automatically arranged.</param>
+        public static void SetAutoLayout(ColumnDefinition element, bool? value)
+        {
+            element.SetValue(AutoLayoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="ColumnDefinition"/></param>
+        /// <returns>A true value indicates this item should be automatically arranged.</returns>
+        public static bool? GetAutoLayout(ColumnDefinition element)
+        {
+            return (bool?)element.GetValue(AutoLayoutProperty);
+        }
+
+        /// <summary>
+        /// Sets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="RowDefinition"/></param>
+        /// <param name="value">A true value indicates this item should be automatically arranged.</param>
+        public static void SetAutoLayout(RowDefinition element, bool? value)
+        {
+            element.SetValue(AutoLayoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="RowDefinition"/></param>
+        /// <returns>A true value indicates this item should be automatically arranged.</returns>
+        public static bool? GetAutoLayout(RowDefinition element)
+        {
+            return (bool?)element.GetValue(AutoLayoutProperty);
+        }
+
+        //// This Property is for ColumnDefinition (DependencyObject) vs. the Grid's FrameworkElement attached property.
+        #pragma warning disable CS0108 // Member hides inherited member; missing new keyword
```

Comment:
Why the need to hide/override the existing Column property?

Reply:
I'm not really overriding/hiding it (thus the warning).  I need my own version of it as I'm adding the property for ColumnDefinition/RowDefinition (which are DependencyObjects) instead of a FrameworkElement from the original Grid.  See the comment above the warning.

Reply:
I did try reusing just the grid property directly and adding the new setter/getter for column/rowdefinition, but then you'll get an AccessViolation.

Reply:
Why would you get an access violation by using the existing `Grid.GetRow` `Grid.SetRow`? Why do we need this at all?

Reply:
The build in Grid Row/Column work on `FrameworkElement`:

    public static int GetRow(FrameworkElement element);

But [ColumnDefinition](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinition) is a `DependencyObject` only.  Therefore I need these extra definition bits so I can do this:

```
            <controls:UniformGrid.ColumnDefinitions>
              <ColumnDefinition controls:UniformGrid.Column="0" Width="0.75*"/>
              <ColumnDefinition controls:UniformGrid.Column="6" Width="0.75*"/>
            </controls:UniformGrid.ColumnDefinitions>
```

It has nothing to do with using Grid.Row/Column for the location of an item in the grid.

We could just make a singular `Index` that works in both cases and wouldn't conflict:

```
    <controls:UniformGrid Rows="5" Columns="5">
            <controls:UniformGrid.ColumnDefinitions>
                    <ColumnDefinition controls:UniformGrid.Index="0" Width="0.5*"/>
                    <ColumnDefinition controls:UniformGrid.Index="4" Width="0.5*"/>
            </controls:UniformGrid.ColumnDefinitions>
            <controls:UniformGrid.RowDefinitions>
                    <RowDefinition controls:UniformGrid.Index="0" Width="0.5*"/>
                    <RowDefinition controls:UniformGrid.Index="4" Width="0.5*"/>
            </controls:UniformGrid.RowDefinitions>
    </controls:UniformGrid>
```

Reply:
But that wouldn't be a uniform grid. A uniform grid is suppose to have the same size column and rows

Reply:
Which brings up the comment I made earlier about the sample being way too complex. The samples should be very simple to get users going. This sample can be simplified by just adding 31 items and binding to the start column and # of rows/columns

Reply:
Agree, the main sample should be much simpler. This is a good sample to have when we have multi sample, so we should still keep it.